From 58111f65c1d006a4a124b1145e464661da723725 Mon Sep 17 00:00:00 2001
From: jamborm <jamborm@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 8 Oct 2015 16:30:06 +0000
Subject: [PATCH] Fix Solaris bootstrap

2015-10-08  Martin Jambor  <mjambor@suse.cz>

	* ipa-cp.c (meet_with_1): Make the argument of abs signed.  Remove
	unnecessary MIN.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@228607 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog | 5 +++++
 gcc/ipa-cp.c  | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index 7ae4f39..d7820e4 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,8 @@
+2015-10-08  Martin Jambor  <mjambor@suse.cz>
+
+	* ipa-cp.c (meet_with_1): Make the argument of abs signed.  Remove
+	unnecessary MIN.
+
 2015-10-08  Yuri Rumyantsev  <ysrumyan@gmail.com>
 
 	* tree-vect-loop.c (vect_analyze_loop_operations): Skip virtual phi
diff --git a/gcc/ipa-cp.c b/gcc/ipa-cp.c
index 0d9fdee..d9d81f1 100644
--- a/gcc/ipa-cp.c
+++ b/gcc/ipa-cp.c
@@ -852,8 +852,8 @@ ipcp_alignment_lattice::meet_with_1 (unsigned new_align, unsigned new_misalign)
     }
   if (misalign != (new_misalign % align))
     {
-      int diff = abs (misalign - (new_misalign % align));
-      align = MIN (align, (unsigned) diff & -diff);
+      int diff = abs ((int) misalign - (int) (new_misalign % align));
+      align = (unsigned) diff & -diff;
       if (align)
 	misalign = misalign % align;
       else
-- 
2.9.2

